plugins/out_stackdriver/stackdriver_operation.c

```diff
@@ -0,0 +1,137 @@
+bool extract_operation(flb_sds_t *operation_id, flb_sds_t *operation_producer, 
+                              bool *operation_first, bool *operation_last, 
+                              msgpack_object *obj, int *extra_subfields)
+{
+    operation_status op_status = NO_OPERATION;
+
+    if (obj->via.map.size != 0) {    	
+        msgpack_object_kv* p = obj->via.map.ptr;
+        msgpack_object_kv* const pend = obj->via.map.ptr + obj->via.map.size;
+
+        for (; p < pend && op_status == NO_OPERATION; ++p) {
+            if (p->val.type == MSGPACK_OBJECT_MAP 
+                && strncmp(OPERATION_FIELD_IN_JSON, p->key.via.str.ptr, p->key.via.str.size) == 0) {
+                
+                op_status = OPERATION_EXISTED;
+                msgpack_object sub_field = p->val;
+                
+                msgpack_object_kv* tmp_p = sub_field.via.map.ptr;
+                msgpack_object_kv* const tmp_pend = sub_field.via.map.ptr + sub_field.via.map.size;
+
+                /* Validate the subfields of operation */
+                for (; tmp_p < tmp_pend; ++tmp_p) {
+                    if (strncmp("id", tmp_p->key.via.str.ptr, tmp_p->key.via.str.size) == 0) {
+                        if(tmp_p->val.type != MSGPACK_OBJECT_STR) {
+                            continue;
+                        }
+                        *operation_id = flb_sds_copy(*operation_id, tmp_p->val.via.str.ptr, tmp_p->val.via.str.size);
+                    }
+                    else if (strncmp("producer", tmp_p->key.via.str.ptr, tmp_p->key.via.str.size) == 0) {
+                        if(tmp_p->val.type != MSGPACK_OBJECT_STR) {
+                            continue;
+                        }
+                        *operation_producer = flb_sds_copy(*operation_producer, tmp_p->val.via.str.ptr, tmp_p->val.via.str.size);
+                    }
+                    else if (strncmp("first", tmp_p->key.via.str.ptr, tmp_p->key.via.str.size) == 0) {
+                        if(tmp_p->val.type != MSGPACK_OBJECT_BOOLEAN) {
+                            continue;
+                        }
+                        *operation_first = tmp_p->val.via.boolean;
+                    }
+                    else if (strncmp("last", tmp_p->key.via.str.ptr, tmp_p->key.via.str.size) == 0) {
+                        if(tmp_p->val.type != MSGPACK_OBJECT_BOOLEAN) {
+                            continue;
+                        }
+                        *operation_last = tmp_p->val.via.boolean;
+                    }
+                    else {
+                        /* extra sub-fields */ 
+                        *extra_subfields += 1;
+                    }
+
+                }
+            }
+        }
+    }
+    
+    return op_status == OPERATION_EXISTED;
+}
+
+void pack_extra_operation_subfields(msgpack_packer *mp_pck, msgpack_object *operation, int extra_subfields) {
+    msgpack_pack_map(mp_pck, extra_subfields);
+
+    msgpack_object_kv* p = operation->via.map.ptr;
```

Comment:
define variables at the beginning of the function:

https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#variable-definitions

Reply:
Done.